pkg/maps/lbmap/bpfservice.go

```diff
@@ -215,5 +241,7 @@ func (l *lbmapCache) prepareUpdate(fe ServiceKey, backends []ServiceValue) *bpfS
 }
 
 func (l *lbmapCache) delete(fe ServiceKey) {
+	l.mutex.Lock()
 	delete(l.entries, fe.String())
+	l.mutex.Unlock()
```

Comment:
same here

Reply:
For single exit functions, there is not much of a point to pay the performance penalty of the defer